### PR TITLE
py-MDAnalysis: update to version 2.0.0

### PR DESCRIPTION
**Pull request by btywoniuk**
* Update to version 2.0.0
* Support for python 3.9
* Patch isn't required any longer

#### Updates of the following port need to be accepted first:
   - py-mmtf-python
   - py-gsd
   - py-GridDataFormats

###### Type(s)

- [ ] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on

macOS 12.0.1 21A559 x86_64
Xcode 13.1 13A1030d

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint`?
- [x] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

###### Package creators:
@MDAnalysis
@richardjgowers

### CODE REVIEW

**Comment by GiovanniBussi** [comment]
I think dependencies should be in alphabetic order why did you change true order? Thanks! Giovanni

File: python/py-MDAnalysis/Portfile [hunk 1]
```diff
@@ -25,28 +25,26 @@ master_sites        pypi:[string index ${python.rootname} 0]/${python.rootname}
...
                             port:py${python.version}-mock \
-                            port:py${python.version}-mmtf-python \
                             port:py${python.version}-networkx \
                             port:py${python.version}-numpy \
                             port:py${python.version}-scipy \
                             port:py${python.version}-six \
                             port:py${python.version}-tqdm
+                            port:py${python.version}-mmtf-python \
+                            port:py${python.version}-GridDataFormats \
+                            port:py${python.version}-gsd \
```
Comment (reneeotten) [inline python/py-MDAnalysis/Portfile:48]: there should be no line continuation for the last dependency here (and yes, please keep them alphabetically sorted)

File: python/py-MDAnalysis/Portfile [hunk 2]
```diff
@@ -4,13 +4,13 @@ PortSystem          1.0
...
 name                py-MDAnalysis
-version             1.0.0
+version             2.0.0
 revision            0
 categories-append   science
 platforms           darwin
 license             GPL-2+
 
-python.versions     27 36 37 38
+python.versions     27 36 37 38 39
```
Comment (IAlibay) [inline python/py-MDAnalysis/Portfile:13]: Just to point out - we got rid of py2.7 as of 2.0.0.

File: python/py-MDAnalysis/Portfile [hunk 3]
```diff
@@ -25,28 +25,26 @@ master_sites        pypi:[string index ${python.rootname} 0]/${python.rootname}
...
-                            port:py${python.version}-GridDataFormats \
-                            port:py${python.version}-gsd \
                             port:py${python.version}-joblib \
                             port:py${python.version}-matplotlib \
                             port:py${python.version}-mock \
-                            port:py${python.version}-mmtf-python \
                             port:py${python.version}-networkx \
                             port:py${python.version}-numpy \
                             port:py${python.version}-scipy \
                             port:py${python.version}-six \
```
Comment (IAlibay) [inline python/py-MDAnalysis/Portfile:44]: six is no longer a core dependency

File: python/py-MDAnalysis/Portfile [hunk 4]
```diff
@@ -25,28 +25,26 @@ master_sites        pypi:[string index ${python.rootname} 0]/${python.rootname}
...
     # OpenMP is disabled now
     # See https://github.com/macports/macports-ports/pull/3902 for an attempt to fix the issue
     build.env-append        MDA_USE_OPENMP=0
     depends_build-append    port:py${python.version}-setuptools
     depends_lib-append      port:py${python.version}-biopython \
-                            port:py${python.version}-GridDataFormats \
-                            port:py${python.version}-gsd \
                             port:py${python.version}-joblib \
                             port:py${python.version}-matplotlib \
                             port:py${python.version}-mock \
```
Comment (IAlibay) [inline python/py-MDAnalysis/Portfile:40]: mock is no longer a core dependency (note this was only used for our testsuite)

**Review by IAlibay — COMMENTED** [review]
Hi,

Sorry to intrude on this PR, just saw this and thought I'd add a couple of quick comments on the 2.0.0 release dependencies.

File: python/py-MDAnalysis/Portfile (hunk 2 above)
Comment (reneeotten) [reply python/py-MDAnalysis/Portfile:13]: @btywoniuk if you want to retain PY27 you'll have to add some additional code within the ` if {${name} ne ${subport}} { ` block:
```
    if  {${python.version} eq 27} {
        version     ????
        revision    0

        checksums   rmd160  ??? \
                    sha256  ??? \
                    size    ???
    }
```

File: python/py-MDAnalysis/Portfile (hunk 1 above)
Comment (btywoniuk) [reply python/py-MDAnalysis/Portfile]: In https://github.com/macports/macports-ports/pull/12970/commits/61130190b0c974a6b9975bd0c6cddacb75f951fa I corrected depends_lib order

File: python/py-MDAnalysis/Portfile (hunk 2 above)
Comment (btywoniuk) [reply python/py-MDAnalysis/Portfile]: In https://github.com/macports/macports-ports/pull/12970/commits/61130190b0c974a6b9975bd0c6cddacb75f951fa  I fully removed python 2.7 from a list of supported versions for this package.

File: python/py-MDAnalysis/Portfile [hunk 5]
```diff
@@ -25,28 +25,24 @@ master_sites        pypi:[string index ${python.rootname} 0]/${python.rootname}
 
 distname            ${python.rootname}-${version}
 
-checksums           rmd160  6ac6c5ad4ad83ff1227ec596cf47de76587d4478 \
-                    sha256  f45a024aca45e390ff1c45ca90beb2180b78881be377e2a1aa9cd6c109bcfa81 \
-                    size    19552874
-
-patchfiles          patch-unistd.diff
-
-if {${name} ne ${subport}} {
```
Comment (reneeotten) [inline python/py-MDAnalysis/Portfile:34]: this `if {${name} ne ${subport}} {` needs to stay

**Review by reneeotten — CHANGES_REQUESTED** [review]
it looks like you accidentally removed the `if`-block, that's not correct so please put that back. When you do that please squash your commits so that there is only one left and update the commit message; then it should be ready-to-go!

File: python/py-MDAnalysis/Portfile (hunk 5 above)
Comment (btywoniuk) [reply python/py-MDAnalysis/Portfile:34]: I squash my commits and restored if {${name} ne ${subport}} {}

**Comment by reneeotten** [comment]
@btywoniuk I just pushed to your branch as you had now a "merge" commit in there, which we don't want (in the future, please do a "rebase" instead of "merge"). Additionally I removed a few redundant fields from the portfile; other than that all your changes are in there. 

Once the CI passes I will merge this - thanks for your contribution!

**Pull request merged**